### DISCUSSION TITLE
Verification vyper file on remix (Contract Verification plugin) will always fail

### DISCUSSION THREAD

**Discussion by Vincent71399** [Cyfrin/moccasin-full-course-cu#14] [Q&A]
I deployed a simple contract (favorites.vy from section 1) to sepolia testnet, then try to verify it using the plugin Contract Verification on remix, however I always got error, 
![image](https://github.com/user-attachments/assets/fe517e05-5105-497b-ab7b-285928fff4d3)
![image](https://github.com/user-attachments/assets/b4051a90-ec38-4964-8809-6217c03ccf37)
![image](https://github.com/user-attachments/assets/c76d731b-d09a-4f96-bf88-9d3225876c84)

I never encounter this when using solidity
Anyone met the same issue?

**Comment by Scofield-Idehen** [comment]
What plugin did you use? 

vyper is not really a native of remix so it uses its own plugin and so you cant really use tools meant for solidity and hope it will work as well 

If you can share more insight on what you were working on, someone could help find a walk around for you. 

best

**Reply by Vincent71399** [reply]
![image](https://github.com/user-attachments/assets/8b7c1e9f-ba22-4655-9bb8-5a07d671a144)
it is called "contract verification"

**Comment by Scofield-Idehen** [comment]
Oh, I see, I do not think vyper is programmed to work with contract verification as etherscan is mostly for solidity-based contracts, I would check around if there is a plugin you can use on remix but for now I suggest you use vyper plugin alone.
